bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java

```diff
@@ -252,54 +269,132 @@ private void scheduleInitialAccess(BoschHttpClient httpClient) {
         }
     }
 
+    /**
+     * Check the bridge access by sending an HTTP request.
+     * Does not throw any exception in case the request fails.
+     */
+    public boolean checkBridgeAccess() {
+        @Nullable
+        BoschHttpClient httpClient = this.httpClient;
+
+        if (httpClient == null) {
+            return false;
+        }
+
+        try {
+            logger.debug("Sending http request to BoschSHC to check access: {}", httpClient);
+            String url = httpClient.getBoschSmartHomeUrl("devices");
+            ContentResponse contentResponse = httpClient.createRequest(url, GET).send();
+
+            // check HTTP status code
+            if (!HttpStatus.getCode(contentResponse.getStatus()).isSuccess()) {
+                logger.debug("Access check failed with status code: {}", contentResponse.getStatus());
+                return false;
+            }
+
+            // Access OK
+            return true;
+        } catch (TimeoutException | ExecutionException | InterruptedException e) {
+            logger.warn("Access check failed because of {}!", e.getMessage());
+            return false;
+        }
+    }
+
     /**
      * Get a list of connected devices from the Smart-Home Controller
      *
      * @throws InterruptedException in case bridge is stopped
      */
-    private boolean getDevices() throws InterruptedException {
+    public List<Device> getDevices() throws InterruptedException {
+        List<Device> emptyDevices = new ArrayList<Device>();
```

Comment:
Use the diamond operator: `new ArrayList<>()`. You can also think about removing this list and use ? `Collections.emptyList()` in the early returns.